### PR TITLE
🌱 E2E: Add build tags

### PR DESCRIPTION
**Pull request by lentzi90**
This makes use of build tags for the e2e tests and related tools. The goal is to make it easier to compile and develop BMO by relaxing the requirement for libvirt bindings. They are now only needed when working on the relevant code.

One exception is golangci-lint. In order for it to go through all code, we need to include the build tags when running the linters. Developers that want to run golangci-lint will need the libvirt bindings.

**What this PR does / why we need it**:

Developer doesn't have libvirt-dev installed and runs `make unit` ->

- Current situation:  ERROR :x: 
- With this PR: Success :heavy_check_mark:

### CODE REVIEW

**Comment by lentzi90** [comment]
Maybe also worth mentioning that at least CAPO and CAPA also uses the `e2e` build tag in this way to keep the e2e code separated from the rest.

**Comment by kashifest** [comment]
This would be useful. Thanks for adding it. 

> One exception is golangci-lint. In order for it to go through all code, we need to include the build tags when running the linters. Developers that want to run golangci-lint will need the libvirt bindings.

Do we need to change something for this? I can see we only run golangci-lint locally, we dont build/tag anything?

**Comment by lentzi90** [comment]
/hold
I see I have missed to change the path in some place. Let me fix that.

**Comment by lentzi90** [comment]
> Do we need to change something for this? I can see we only run golangci-lint locally, we dont build/tag anything?

If we want golangci-lint to check the code that binds to libvirt there is no way around it unfortunately. I'm not sure how golangci-lint works, but in some sense it must be building the code because it sees the same error as `go build` does if the libvirt-dev package isn't installed.

We could chose to split the linting so that `test/vbmctl` would be a separate target. Then it would be possible to run `make lint` without libvirt bindings. However, then there is the risk that we forget to run the separate target. I don't have a strong opinion on what is better.

We could for example have it like this:

- `make lint`: Only checks code without build tags. Works without libvirt-dev.
- `make lint-e2e`: Only checks code under `test`. Requires libvirt-dev.

File: Makefile [hunk 1]
```diff
@@ -198,7 +198,11 @@ deploy-cli:
 
 .PHONY: build-e2e
 build-e2e:
-	cd test; go build ./...
+	cd test; go build --tags=e2e ./...
+
+.PHONY: build-vbmctl
+build-vbmctl:
+	cd test; go build --tags=e2e,vbmctl -ldflags $(LDFLAGS) -o $(abspath $(BIN_DIR)/vbmctl) ./vbmctl/main.go
```
[the remark this answers is not in the document]
Comment (lentzi90) [reply Makefile:205]: I think it is better to leave it. If we add it htere, then libvirt-dev would be needed for `make build`. We will anyway build this code as part of the e2e pipeline.

[the remark this answers is not in the document]
Comment (lentzi90) [reply Makefile:205]: Hmm now I wonder if we should do the same for golanci-lint also. It is not so nice to have them behave differently.

- `make lint` -> lint everything
- `make build` -> build everything except vbmctl

**Comment by lentzi90** [comment]
These fixture test failures are driving me mad... Attempting to fix in https://github.com/metal3-io/baremetal-operator/pull/2043

**Comment by mquhuy** [comment]
> > Do we need to change something for this? I can see we only run golangci-lint locally, we dont build/tag anything?
> 
> If we want golangci-lint to check the code that binds to libvirt there is no way around it unfortunately. I'm not sure how golangci-lint works, but in some sense it must be building the code because it sees the same error as `go build` does if the libvirt-dev package isn't installed.
> 
> We could chose to split the linting so that `test/vbmctl` would be a separate target. Then it would be possible to run `make lint` without libvirt bindings. However, then there is the risk that we forget to run the separate target. I don't have a strong opinion on what is better.
> 
> We could for example have it like this:
> 
> * `make lint`: Only checks code without build tags. Works without libvirt-dev.
> * `make lint-e2e`: Only checks code under `test`. Requires libvirt-dev.

No strong preference from me, but imo it would be good to not have to install libvirt-dev in the lint job.

We could repurpose the current lint workflow to be the lint-e2e, and config it so that it only runs when we edit the code inside the `test/vbmctl` directory.

Otherwise, I think this PR is a nice improvement. Not sure if any developer will ever work on this repo without installing `libvirt-dev`, but it's nice to at least have options.

**Comment by lentzi90** [comment]
/hold cancel
Tests are passing :tada: 

I'm still open to discuss how we handle the linters, but if you are happy like this, then lets merge @kashifest

**Comment by kashifest** [comment]
Lets merge it and see if the we need a change in the linter once we hit an issue
/approve

**Comment by lentzi90** [comment]
We seem to be hitting the golanci-lint deadline. Fixing here: https://github.com/metal3-io/baremetal-operator/pull/2046

**Pull request merged**